optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/anchor/AnchorVariableDemand.java

```diff
@@ -68,7 +68,7 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return CLASS_NAME_HASH_CODE + sourceVariableDescriptor.hashCode();
+        return Objects.hash(this.getClass().getName(), sourceVariableDescriptor);
```

Comment:
Change breaks equals contract, because equality doesn't take into account subclass names.
To avoid that, use Objects.hash(AnchorVariableDemand.class.getName(), ...)